web/src/app/core/services/authentication.service.ts

```diff
@@ -144,6 +149,16 @@ export class AuthenticationService {
             });
     }
 
+    // This function will set the progress bar status
+    public setProgressBar(status: boolean): void {
+       this.spinnerSubject.next(status);
+    }
+
+    // this function will return the status of progress bar to main component
+    public getProgressBar(): Observable<Boolean> {
```

Comment:
Both these functions to be moved to its own service as mentioned here https://github.com/DashboardHub/PipelineDashboard/pull/1222/files#r282212369